cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/UserPermissionsParameters.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet.parameters;
+
+import com.linkedin.kafka.cruisecontrol.servlet.CruiseControlEndPoint;
+import java.io.UnsupportedEncodingException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import static com.linkedin.kafka.cruisecontrol.servlet.parameters.ParameterUtils.REASON_PARAM;
+
+
+/**
+ * Parameters for {@link CruiseControlEndPoint#PERMISSIONS}
+ *
+ * <pre>
+ *    GET /kafkacruisecontrol/permissions?json=[true/false]&amp;
+ *    doAs=[user]&amp;get_response_schema=[true/false]&amp;reason=[reason-for-request]
+ * </pre>
+ */
+public class UserPermissionsParameters extends AbstractParameters {
+    protected static final SortedSet<String> CASE_INSENSITIVE_PARAMETER_NAMES;
+    private static final String ANONYMOUS = "ANONYMOUS";
+    static {
+        SortedSet<String> validParameterNames = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+        validParameterNames.add(REASON_PARAM);
+        validParameterNames.addAll(AbstractParameters.CASE_INSENSITIVE_PARAMETER_NAMES);
+        CASE_INSENSITIVE_PARAMETER_NAMES = Collections.unmodifiableSortedSet(validParameterNames);
```

Comment:
Just curious, why does it have to be sorted?

Reply:
It is done analogously with other Parameters classes, e.g. AdminParameters.java, BootstrapParameters.java etc